src/main/java/uk/sky/cirrus/locking/CassandraLockingMechanism.java

```diff
@@ -0,0 +1,126 @@
+     */
+    @Override
+    public boolean acquire() throws CannotAcquireLockException {
+        try {
+            ResultSet resultSet = session.execute(insertLockQuery.bind(lockName, clientId));
+            Row currentLock = resultSet.one();
+            if (currentLock.getBool("[applied]") || clientId.equals(currentLock.getString("client"))) {
+                return true;
+            } else {
+                log.info("Lock currently held by {}", currentLock);
+                return false;
+            }
+        } catch (WriteTimeoutException wte) {
+            log.warn("Query to acquire lock for {} failed to execute: {}", clientId, wte.getMessage());
+            return false;
+        } catch (DriverException de) {
+            throw new CannotAcquireLockException(String.format("Query to acquire lock %s for client %s failed to execute", lockName, clientId), de);
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Deletes lock from locks table.
+     * Will retry to release lock if WriteTimeoutException thrown.
+     * If a WriteTimeoutException has previously been thrown this
+     * will check if the lock did actually successfully deleted.
+     *
+     * @throws CannotAcquireLockException if any DriverException thrown while executing queries.
+     */
+    @Override
+    public void release() throws CannotReleaseLockException {
+        while (true) {
+            try {
+                ResultSet resultSet = session.execute(deleteLockQuery.bind(lockName, clientId));
+                Row result = resultSet.one();
+
+                if (result.getBool("[applied]") || !result.getColumnDefinitions().contains("client")) {
+                    log.info("Lock released for {} by client {} at: {}", lockName, clientId, System.currentTimeMillis());
+                    return;
+                }
+
+                String clientReleasingLock = result.getString("client");
+                if (!clientReleasingLock.equals(clientId)) {
+                    if (isRetryAfterWriteTimeout) {
+                        log.info("Released lock for client {} in retry attempt after WriteTimeoutException", clientReleasingLock);
+                        return;
+                    } else {
+                        throw new CannotReleaseLockException(
+                                String.format("Lock %s attempted to be released by a non lock holder (%s). Current lock holder: %s", lockName, clientId, clientReleasingLock));
+                    }
+                }
+
+            } catch (WriteTimeoutException e) {
+                isRetryAfterWriteTimeout = true;
+                waitToRetryRelease();
+            } catch (DriverException e) {
+                log.error("Query to release lock failed to execute for {} by client {}", lockName, clientId, e);
+                throw new CannotReleaseLockException("Query failed to execute", e);
+            }
+        }
+    }
+
+    private void waitToRetryRelease() {
```

Comment:
To be honest I don't really get why we have this single retry mechanism. Why not do it for acquire as well? Why not just increase the write timeouts?